### PR TITLE
feat(om): make Event Hubs stream token converter registration public

### LINKED ISSUE #184: om: StreamSequenceTokenJsonConverters.Register throws on duplicate registration, making converter setup order-dependent
## Summary

`StreamSequenceTokenJsonConverters.Register` throws `InvalidOperationException` when a type descriptor is already registered. Because the registry is process-wide and `UseEnrichedDataAdapter()` registers into it eagerly, any application that also registers the Event Hub token converters itself becomes order-dependent, and the losing registrar throws at silo startup.

Found while migrating Clever Pricing Engine (OM's first consumer) to 0.5.6.

## Why an application registers them at all

`StreamCursor` and `MessageTracker` persist stream sequence tokens as JSON inside grain state. Anything that deserializes that state needs the Event Hub token converters registered — including processes that never configure an Event Hub stream provider:

- the test fixture, which uses in-memory grain storage with the production `JsonSerializerOptions`;
- a console tool that reads grain state blobs offline;
- a background archiver job.

None of those call `UseEnrichedDataAdapter()`, so they must call `StreamSequenceTokenJsonConverters.Register` directly. There is no public equivalent of the internal `EventHubStreamSequenceTokenJsonConverters.Register()`, so the three registrations get copied into the application:

```csharp
StreamSequenceTokenJsonConverters.Register(
    "orleans.event-hubs.sequence-token", new EventHubSequenceTokenJsonConverter());
StreamSequenceTokenJsonConverters.Register(
    "orleans.event-hubs.sequence-token-v2", new EventHubSequenceTokenV2JsonConverter());
StreamSequenceTokenJsonConverters.Register(
    EnrichedEventHubSequenceToken.TypeAlias, new EnrichedEventHubSequenceTokenJsonConverter());
```

## The collision

The silo *also* calls `UseEnrichedDataAdapter()`, which calls `EventHubStreamSequenceTokenJsonConverters.Register()` and registers the same three descriptors, with no guard of its own. Two registrars, same three keys, and whoever runs second throws.

In our silo it happens to work, purely by ordering: `UseEnrichedDataAdapter()` runs eagerly during `AddEventHubStreams(...)`, while the static constructor holding the application's copy is first triggered ~170 lines later by an eagerly constructed singleton. So OM wins, the application's first `Register` throws, and a `try`/`catch (InvalidOperationException)` wrapped around all three swallows it — silently skipping registrations two and three as well. Correct end state, entirely by accident.

Reverse the order — resolve the serializer options earlier, or reorder silo configuration — and `UseEnrichedDataAdapter()` throws an unhandled `InvalidOperationException` during silo startup. No test catches it, because a test fixture that does not configure Event Hub streams never creates the collision.

## Suggested fix

Any of these would remove the order dependence:

1. **`TryRegister`** returning `bool`, so a second registrar can be explicit about tolerating a duplicate.
2. **Make `Register` idempotent for an identical re-registration** — same descriptor, same converter type — and keep throwing only on a genuine conflict where two different converters claim one descriptor. That is the conflict worth protecting against; re-registering the library's own converter is not.
3. **Make `EventHubStreamSequenceTokenJsonConverters` public**, so applications that need the converters outside a silo can call the same entry point instead of copying the three lines. Best combined with 1 or 2, since it does not by itself stop `UseEnrichedDataAdapter()` from colliding with an earlier call.

Option 2 plus 3 is what we would have wanted: one public, idempotent call, usable from a silo, a test fixture, or an offline tool, with no ordering rules to reason about.

## Impact

Low severity today — it needs a specific ordering to bite — but it is a latent startup crash that is invisible to tests, and the workaround (`try`/`catch` around a registration block) is the kind of thing that hides real conflicts.

### PR DESCRIPTION
**Pull request by egil**
Closes #184.

## What the issue reported, and what is actually true

Issue #184 reports that `StreamSequenceTokenJsonConverters.Register` throws on any duplicate registration, making converter setup order-dependent and leaving a latent silo-startup crash.

That premise does not hold. `Register` has been idempotent for an identical re-registration since the feature's first commit (`739785a`, contained in every release including `v0.5.6`): same descriptor, same token type and same converter type returns silently, and it throws only when a *different* converter claims a descriptor.

I reproduced the issue's exact scenario against unmodified `main` — the application's three copied `Register` lines first, then `UseEnrichedDataAdapter()` second, the order the issue predicts crashes startup — and it passes without throwing. So the consumer-side `try`/`catch (InvalidOperationException)` is dead code, and it would mask a genuine conflict if one ever arose.

## What was real, and is fixed here

Suggestion 3 in the issue. `EventHubStreamSequenceTokenJsonConverters` was `internal` with exactly one call site, so any process that reads the same persisted grain state without configuring an Event Hub stream provider had to copy three descriptor strings and three converter constructions — a test fixture on in-memory storage using the production `JsonSerializerOptions`, a tool reading grain state blobs offline, a background archiver. This repo's own EventHubs test project contained precisely that copy.

- `EventHubStreamSequenceTokenJsonConverters` is now public, with `Register()` and the two descriptors exposed as `EventHubSequenceTokenTypeDescriptor` / `EventHubSequenceTokenV2TypeDescriptor`, so no string has to be copied.
- New `services.AddEventHubStreamSequenceTokenJsonConverters()` for callers with a container, mirroring the core package's `AddStreamSequenceTokenJsonConverter<TToken, TConverter>` style.
- New `StreamSequenceTokenJsonConverters.TryRegister(...)` for callers that want to report a conflict rather than catch one. It returns `true` when the registry holds an equivalent converter once the call returns — whether it added one or found one — and `false` when a different converter owns the descriptor. Deliberately not `TryAdd` semantics: the caller's question is "is my converter in place?", not "did I mutate the registry?".
- The test project's copy of the three registration lines is replaced by the public call.

## Tests

12 new tests. The core registry gains coverage for the idempotent path, which nothing pinned before, plus each `TryRegister` outcome. The EventHubs project gains round-trips of every Event Hub token type through `StreamCursor`, repeated registration, registering before `UseEnrichedDataAdapter()`, and — new — an assertion that `UseEnrichedDataAdapter()` registers the converters at all, which nothing checked previously.

The two ordering tests pass on unmodified `main`; they pin behaviour that already works rather than fixing a break, and are included so it cannot silently regress.

One note on assertion strength: Orleans' `EventSequenceToken.Equals` compares only `SequenceNumber` and `EventIndex`, so a naive `Assert.Equal` on two tokens passes even when a round trip downcast an enriched token to a bare one — the exact failure this registry exists to prevent. The tests assert concrete type and every carried field instead.

## Docs

`README.md` gains a section on registering the converters outside a silo, and states that wrapping registration in `try`/`catch (InvalidOperationException)` hides the only conflict worth knowing about.

`api-design.md` records the idempotency contract and, separately, the read/write asymmetry for descriptor aliases: one token type may hold several descriptors and every registered descriptor decodes, which is how a descriptor can be renamed while previously persisted state stays readable — but the first registration to claim a token type owns its persisted `Kind`, so a later alias is decode-only. That is the opposite of `IServiceProvider`, where the last registration wins and is how a caller overrides a default. There is no override here by design: the winner is persisted rather than resolved, so letting a late alias change the wire format would let registration order, which no composition root controls, silently rewrite durable state.

## Verification

`dotnet build Egil.Orleans.Messaging/Egil.Orleans.Messaging.slnx -c Release` — 0 warnings, 0 errors.
`dotnet test Egil.Orleans.Messaging/Egil.Orleans.Messaging.slnx` — 389 passed, 0 failed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Egil.Orleans.Messaging/test/Egil.Orleans.Messaging.Streams.EventHubs.Tests/EventHubs/EnrichedEventHubAdapterExtensionsTests.cs [hunk 1]
```diff
@@ -22,17 +25,26 @@ public void UseEnrichedDataAdapter_registers_configuration_callback()
...
+        configurator.UseEnrichedDataAdapter();
 
-        public Action<Action<IServiceCollection>> ConfigureDelegate => configure =>
-        {
-            ConfigureCalls++;
-            LastConfigureAction = configure;
-        };
+        var json = JsonSerializer.Serialize(new StreamCursor("orders", token));
+        var restored = JsonSerializer.Deserialize<StreamCursor>(json)!.Token;
+        var enriched = Assert.IsType<EnrichedEventHubSequenceToken>(restored);
```
[the remark this answers is not in the document]
Comment (egil) [reply Egil.Orleans.Messaging/test/Egil.Orleans.Messaging.Streams.EventHubs.Tests/EventHubs/EnrichedEventHubAdapterExtensionsTests.cs:52]: Correct, and fixed in f6fbbaf by renaming rather than by adding a reset. Renamed to `Enriched_token_round_trips_after_UseEnrichedDataAdapter`, which is what it actually verifies, with a comment recording why causation is untestable here. A reset would contradict a documented non-goal (a persisted `Kind` must keep decoding the same way for the life of the process) and would race under xUnit parallel class execution; process isolation would need a dedicated single-test assembly per case, since the assembly is the process boundary. The core registry tests in `StreamSequenceTokenRegistrationTests` are properly isolated — they claim a fresh GUID descriptor per test.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: Egil.Orleans.Messaging/test/Egil.Orleans.Messaging.Streams.EventHubs.Tests/EventHubs/EventHubStreamSequenceTokenJsonConvertersTests.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    [Fact]
+    public void AddEventHubStreamSequenceTokenJsonConverters_registers_the_converters()
+    {
+        var services = new ServiceCollection();
+
+        var returned = services.AddEventHubStreamSequenceTokenJsonConverters();
+
+        Assert.Same(services, returned);
+        AssertRoundTrips(CreateEnrichedToken());
```
[the remark this answers is not in the document]
Comment (egil) [reply Egil.Orleans.Messaging/test/Egil.Orleans.Messaging.Streams.EventHubs.Tests/EventHubs/EventHubStreamSequenceTokenJsonConvertersTests.cs:61]: Agreed, same root cause. Fixed in f6fbbaf: renamed to `AddEventHubStreamSequenceTokenJsonConverters_returns_the_service_collection`, since the fluent return is the part this test genuinely isolates, and the round trip is now commented as a smoke check that the extension leaves the registry usable. `Register_registers_every_event_hub_token_type` had the same overclaiming name and is now `Every_event_hub_token_type_round_trips`. The two no-throw tests (`Register_is_idempotent_across_repeated_calls`, `Register_before_UseEnrichedDataAdapter_does_not_throw`) are unaffected — prior registry state does not weaken a no-throw assertion, it makes it more representative.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
